src/main.cpp

```diff
@@ -86,8 +85,17 @@ void setup () {
   }
   nvs.close();
 
+  PyranometerSensor ps;
+  data = (uint8_t*)malloc(8 * sizeof(uint8_t));
```

Comment:
This looks great, but it should use the new data buffer instead of `malloc`/`free` and the whole block should execute before we access non-volatile storage.